wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ConversationListViewController+UISearchResultsUpdating.swift

```diff
@@ -0,0 +1,35 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+
+private let searchDebounceInterval: TimeInterval = 0.2
+
+extension ConversationListViewController: UISearchResultsUpdating {
+
+    func updateSearchResults(for searchController: UISearchController) {
+
+        NSObject.cancelPreviousPerformRequests(
+            withTarget: self,
+            selector: #selector(applySearchText),
+            object: nil
+        )
+
+        perform(#selector(applySearchText), with: nil, afterDelay: searchDebounceInterval)
```

Comment:
Should we go to prefer Combine here over Objective-C sectors?

Reply:
Since we are doing it like this before and it comes for free with `NSObject` I prefer to keep it that way.